examples/chip-tool/commands/common/Command.cpp

```diff
@@ -330,28 +351,41 @@ bool Command::InitArgument(size_t argIndex, char * argValue)
         break;
     }
 
-    case ArgumentType::Bool:
-    case ArgumentType::Number_uint8: {
-        isValidArgument = HandleNullableOptional<uint8_t>(arg, argValue, [&](auto * value) {
-            // stringstream treats uint8_t as char, which is not what we want here.
-            uint16_t tmpValue;
-            std::stringstream ss;
-            isHexNotation ? ss << std::hex << argValue : ss << argValue;
-            ss >> tmpValue;
-            if (chip::CanCastTo<uint8_t>(tmpValue))
+    case ArgumentType::Bool: {
+        isValidArgument = HandleNullableOptional<bool>(arg, argValue, [&](auto * value) {
+            // Start with checking for actual boolean values.
+            if (strcmp(argValue, "true") == 0)
             {
-                *value = static_cast<uint8_t>(tmpValue);
+                *value = true;
+                return true;
+            }
 
-                uint64_t min = chip::CanCastTo<uint64_t>(arg.min) ? static_cast<uint64_t>(arg.min) : 0;
-                uint64_t max = arg.max;
-                return (!ss.fail() && ss.eof() && *value >= min && *value <= max);
+            if (strcmp(argValue, "false") == 0)
```

Comment:
Suggest std::string utility to do case-insensitive compare, so that case exactness on true/false not needed, and prettier code as well

Reply:
@tcarmelveilleux Do we have such a utility already?

That said, I could `strcasecmp` here if we just want case-insensitivity....

Reply:
Did strcasecmp.